envoy/config/bootstrap/v2/bootstrap.proto

```diff
@@ -166,6 +166,10 @@ message ClusterManager {
   // <envoy_api_field_core.ApiConfigSource.api_type>` :ref:`GRPC
   // <envoy_api_enum_value_core.ApiConfigSource.ApiType.GRPC>`.
   envoy.api.v2.core.ApiConfigSource load_stats_config = 4;
+
+  // Optional options for upstream connections.
+  // This may be overridden on a per-cluster basis by upstream_connection_options in the cds_config.
+  envoy.api.v2.UpstreamConnectionOptions upstream_connection_options = 5;
```

Comment:
It is preferable to omit this. Envoy doesn't in general do hierarchical config, in the sense of setting an option at a higher level and overriding later.

This has been handled inconsistently (for instance bind config). But unless there's a compelling reason, please remove this (and "bool disable = 4" from the TcpKeepalive message).

Reply:
OK. I don't have a particularly good excuse for wanting this, and removing it will simplify the code a little, so I'll remove it.